clmm/support/mock_data.py

```diff
@@ -486,7 +490,7 @@ def _generate_galaxy_catalog(
     return galaxy_catalog[cols]
 
 
-def _draw_source_redshifts(zsrc, zsrc_min, zsrc_max, ngals):
+def _draw_source_redshifts(zsrc, zsrc_min, zsrc_max, ngals, coordinate_system):
```

Comment:
It's a bit counter intuitive that a functions that only draw redshifts needs to know about the coordinate system. I understand because this function is called first to generate the mock data and the initial galaxy catalog comes from it.

An option would be to declare an empty galaxy catalog GCData with the coord system in the metadata, and then, have `_draw_source_redhsifts` adds to that GCData. Similarly to what `_draw_galaxy_positions` or `_compute_photoz_pdfs` do?

In any case, if we keep it like this, `coordinate_system` needs an argument docstring. It's currently missing.